### DISCUSSION TITLE
Login to the OIDC app from the Okta dashboard

### DISCUSSION THREAD

**Discussion by rajubushanam** [nextauthjs/next-auth#5194] [Help]
### Working Scenario
We are utilizing `next-auth` and Okta for our app.
We are able to currently login a user into our app through Okta OIDC by calling `signIn()` function from next-auth when user clicks on Sign in button on our app's login page.

### Our goal is:
1. A user is logged into Okta already from a different tab by visiting okta dasboard url (ex: `https://test.okta.com`).
2. Because we have configured our OIDC app to show on user's okta dashboard, they should be able to click on the app icon and get redirected to our app, in this case we are redirecting it to - `https://ourdomain/api/auth/callback/okta`.
3. After it redirects here we were expecting that next-auth will make a call to Okta and make sure there is a valid login session and authenticates the user.
But we are seeing the following error:
```
TypeError: id_token not present in TokenSet
    at TokenSet.claims (/app/node_modules/openid-client/lib/token_set.js:23:13)
    at oAuthCallback (/app/node_modules/next-auth/core/lib/oauth/callback.js:132:24)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Object.callback (/app/node_modules/next-auth/core/routes/callback.js:50:11)
    at async NextAuthHandler (/app/node_modules/next-auth/core/index.js:186:28)
    at async NextAuthNextHandler (/app/node_modules/next-auth/next/index.js:23:19)
    at async /app/node_modules/next-auth/next/index.js:59:32
    at async Object.apiResolver (/app/node_modules/next/dist/server/api-utils/node.js:179:9)
    at async NextNodeServer.runApi (/app/node_modules/next/dist/server/next-server.js:381:9)
    at async Object.fn (/app/node_modules/next/dist/server/base-server.js:497:37)`
```

### Okta Configuration
On Okta we have:
1. OIDC app with clientId and clientSecret.
5. Custom Authorization server that mints the token. This was needed to connect an inline hook to our service to append custom payload to the token.
6. Issuer on the custom authorization server will be slightly different from the org issuer in OIDC app.

On our next-auth initialization with Okta Provider we are passing the following:
```
providers: [
    OktaProvider({
      clientId: {{OIDC_app_clientId}},
      clientSecret: {{OIDC_app_clientSecret}},
      issuer: 'https://test.okta.com',
      authorization: `${CUSTOM_AUTH_SERVER_URL}/v1/authorize`,
      token: `${CUSTOM_AUTH_SERVER_URL}/v1/token`,
    }),
  ]
```

CUSTOM_AUTH_SERVER_URL = 'https://test.okta.com/oauth2/janskjbdbadl'

Any help is greatly appreciated 🙏

**Comment by LinnJS** [comment]
I am not 100% sure this is the answer but I would assume when you click the link in the Okta dashboard there is a token they are trying to pass to the app. So maybe you can use one of the various NextAuth callback functions to take the token from the URL or the headers and give it to NextAuth to pass back to Okta to complete the round trip.

Sounds like the hand off from the Okta dashboard to your app is getting fumbled. 

Callback docs: https://next-auth.js.org/configuration/callbacks
